src/util/templates/EjsTemplateEngine.ts

```diff
@@ -1,35 +1,25 @@
-/* eslint-disable tsdoc/syntax */
-// tsdoc/syntax cannot handle `@range`
-import type { TemplateFunction } from 'ejs';
-import { compile, render } from 'ejs';
-import type { TemplateEngine, Template } from './TemplateEngine';
-import { getTemplateFilePath, readTemplate } from './TemplateEngine';
+import { render } from 'ejs';
+import { ExtensionBasedTemplateEngine } from './ExtensionBasedTemplateEngine';
+import type { TemplateEngineInput } from './TemplateEngine';
+import { getTemplateFilePath, readTemplate } from './TemplateUtil';
 import Dict = NodeJS.Dict;
 
 /**
  * Fills in EJS templates.
  */
-export class EjsTemplateEngine<T extends Dict<any> = Dict<any>> implements TemplateEngine<T> {
-  private readonly applyTemplate: Promise<TemplateFunction>;
+export class EjsTemplateEngine<T extends Dict<any> = Dict<any>> extends ExtensionBasedTemplateEngine<T> {
   private readonly baseUrl: string;
 
   /**
    * @param baseUrl - Base URL of the server.
-   * @param template - The default template @range {json}
    */
-  public constructor(baseUrl: string, template?: Template) {
-    // EJS requires the `filename` parameter to be able to include partial templates
-    const filename = getTemplateFilePath(template);
+  public constructor(baseUrl: string) {
+    super([ 'ejs' ]);
```

Comment:
For both ejs and hbs, make this an optional constructor parameter that defaults to the expected value. Just in case someone has a set of hbs/ejs templates with a different extensions.